core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -317,10 +317,6 @@ protected Node visitShowRoles(ShowRoles node, Void context)
         @Override
         protected Node visitShowRoleGrants(ShowRoleGrants node, Void context)
         {
-            if (node.getCatalog().isEmpty() && session.getCatalog().isEmpty()) {
-                throw semanticException(MISSING_CATALOG_NAME, node, "Catalog must be specified when session catalog is not set");
```

Comment:
This should come with a test addition/change. Seems we're missing testing coverage.